packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/WalletConnectProposalModal.tsx

```diff
@@ -0,0 +1,95 @@
+import {
+    selectPendingProposal,
+    sessionProposalApproveThunk,
+    sessionProposalRejectThunk,
+} from '@suite-common/walletconnect';
+import { Banner, H2, NewModal, Note, Paragraph } from '@trezor/components';
+import { spacings } from '@trezor/theme';
+
+import { onCancel } from 'src/actions/suite/modalActions';
+import { Translation } from 'src/components/suite';
+import { useDispatch, useSelector } from 'src/hooks/suite';
+
+interface WalletConnectProposalModalProps {
+    eventId: number;
+}
+
+export const WalletConnectProposalModal = ({ eventId }: WalletConnectProposalModalProps) => {
+    const dispatch = useDispatch();
+    const pendingProposal = useSelector(selectPendingProposal);
+
+    const handleAccept = () => {
+        dispatch(sessionProposalApproveThunk({ eventId }));
+        dispatch(onCancel());
+    };
+    const handleReject = () => {
+        dispatch(sessionProposalRejectThunk({ eventId }));
+        dispatch(onCancel());
+    };
+
+    return (
+        <NewModal
+            onCancel={handleReject}
+            iconName="plugs"
+            variant="primary"
+            bottomContent={
+                <>
+                    <NewModal.Button variant="tertiary" onClick={handleReject}>
+                        <Translation id="TR_CANCEL" />
+                    </NewModal.Button>
+                    <NewModal.Button
+                        variant="primary"
+                        onClick={handleAccept}
+                        isDisabled={
+                            !pendingProposal || pendingProposal.expired || pendingProposal.isScam
+                        }
+                    >
+                        <Translation id="TR_CONFIRM" />
+                    </NewModal.Button>
+                </>
+            }
+            heading={<Translation id="TR_TREZOR_CONNECT" />}
+        >
+            <H2>{pendingProposal?.params.proposer.metadata.name}</H2>
```

Comment:
isn't it maybe better to do some early return to get rid of this optional chaining everywhere?